pallets/price-oracle/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+// This file is part of Basilisk-node.
+
+// Copyright (C) 2020-2021  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet_prelude::Weight;
+use frame_support::sp_runtime::traits::Zero;
+use primitives::{
+	asset::AssetPair,
+	traits::{AMMHandlers, AMMTransfer},
+	AssetId, Balance,
+};
+use sp_std::marker::PhantomData;
+use sp_std::prelude::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+mod types;
+pub use types::*;
+
+pub mod weights;
+use weights::WeightInfo;
+
+mod benchmarking; // TODO: rebenchmark
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+pub type AssetPairId = Vec<u8>;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Weight information for the extrinsics.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Calculation error occurred while calculating average price
+		PriceComputationError,
+
+		/// An unexpected overflow occurred
+		UpdateDataOverflow,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Pool was registered. [AssetPair]
+		PoolRegistered(AssetPair),
+	}
+
+	/// The number of assets registered and handled by this pallet.
+	#[pallet::storage]
+	#[pallet::getter(fn num_of_assets)]
+	pub type NumOfTrackedAssets<T: Config> = StorageValue<_, u32, ValueQuery>;
```

Comment:
`TrackedAssetsCount` . maybe?

And it can be AssetId type instead of u32

Reply:
We use `u32` because we use the value here https://github.com/galacticcouncil/Basilisk-node/blob/534465213edee32e2a9ad24466bd1ebf1685b95a/pallets/price-oracle/src/lib.rs#L119 and autogenerated parametrized weight functions uses `u32`https://github.com/galacticcouncil/Basilisk-node/blob/534465213edee32e2a9ad24466bd1ebf1685b95a/pallets/price-oracle/src/weights.rs#L82